addons/account/views/report_payment_receipt_templates.xml

```diff
@@ -33,47 +33,80 @@
+                                        <t t-set="otherCurrency" t-value="True"/>
+                                    </t>
+                                </t>
+                                <!-- Add a column if there are different currencies -->
+                                <t t-if="otherCurrency == True">
+                                    <th class="text-right"><span>Amount In Currency</span></th>
+                                </t>
+                                <th class="text-right"><span>Amount</span></th>
+                            </tr>
+                        </thead>
+                        <tbody>
                             <!-- MOVE -->
                             <t t-if="inv.move_type != 'entry'">
                                 <tr>
                                     <td><span t-field="inv.invoice_date"/></td>
                                     <td><span t-field="inv.name"/></td>
                                     <td><span t-field="inv.ref"/></td>
+                                    <t t-if="otherCurrency == True">
+                                        <td/>
+                                    </t>
                                     <td class="text-right"><span t-field="inv.amount_total"/></td>
                                 </tr>
                                 <!-- PAYMENTS/REVERSALS -->
                                 <tr t-foreach="inv._get_reconciled_invoices_partials()" t-as="par">
-                                    <t t-set="amount" t-value="par[1]"/>
                                     <t t-set="payment" t-value="par[2].move_id"/>
                                     <td><span t-field="payment.date"/></td>
                                     <td><span t-field="payment.name"/></td>
                                     <td><span t-field="payment.ref"/></td>
-                                    <td class="text-right">-<span t-esc="amount" t-options="{'widget': 'monetary', 'display_currency': o.currency_id}"/></td>
+                                    <t t-set="amountPayment" t-value="-payment.amount_total"/>
+                                    <t t-set="amountInvoice" t-value="-par[1]"/>
+                                    <t t-set="currencyPayment" t-value="payment.currency_id"/>
+                                    <!-- Fill the column "Amount In Currency" only if necessary -->
+                                    <t t-if="otherCurrency == True">
+                                        <t t-if="not currencyPayment == currencyInvoice">
+                                            <td class="text-right"><span t-esc="amountPayment" t-options="{'widget': 'monetary', 'display_currency': currencyPayment}"/></td>
+                                        </t>
+                                        <t t-else="">
+                                            <td/>
+                                        </t>
+                                    </t>
```

Comment:
same as [1], may be simplified in
`<td t-if="otherCurrency" class="text-right"><span t-if="currencyPayment != currencyInvoice" t-esc="amountPayment" t-options="{'widget': 'monetary', 'display_currency': currencyPayment}"/></td>`